src/6_branch.js

```diff
@@ -541,6 +541,56 @@ Branch.prototype['init'] = wrap(
 );
 
 /**
+ * @function Branch.deepview
+ * @param {function(?Error, utils.sessionData=)=} callback - _optional_ - callback to read the session data.
+ */
+Branch.prototype['deepview'] = wrap(
+	callback_params.CALLBACK_ERR_DATA,
+	function(done, data, options) {
+		var self = this;
+
+		if (!options) {
+			options = { };
+		}
+		var cleanedData = utils.cleanLinkData(data);
+
+		if (cleanedData['tags']) {
+			cleanedData['tags'] = goog.json.serialize(cleanedData['tags']);
+		}
+
+		if (cleanedData['data']) {
+			cleanedData['metadata'] = cleanedData['data'];
+			delete cleanedData['data'];
+		}
+
+		if (options['open_app']) {
+			cleanedData['open_app'] = true;
+		}
+
+		var referringLink = self._referringLink();
+		if (referringLink && !options['make_new_link']) {
+			cleanedData['link_click_id'] = referringLink.substring(
+				referringLink.lastIndexOf('/') + 1, referringLink.length
+			);
+		}
+
+		this._api(resources.deepview, cleanedData, function(err, data) {
+			if (err) {
+				return done(err);
+			}
+
+			if (typeof data === 'function') {
+				Branch.prototype['deepviewCta'] = data;
+			}
```

Comment:
no we don't. because this will cause two branch.deepview callbacks. 

for example, if the user does `branch.deepview(my_data, my_callback);` `my_callback` will be called twice.